shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -187,7 +187,13 @@
     const flutter::EmbeddedViewParams& params) {
   // TODO(amirh): assert that this is running on the platform thread once we support the iOS
   // embedded views thread configuration.
-  // TODO(amirh): do nothing if the params didn't change.
+
+  // Do nothing if the params didn't change.
+  if (composite_params_.count(view_id) == 1 && composite_params_[view_id] == params) {
+    return picture_recorders_[view_id]->getRecordingCanvas();
+  }
+  composite_params_[view_id] = params;
+
```

Comment:
Do we need to call `setFrame` after a UIView is removed from the super view and then added again? (if so we should probably clear the value in `DetachUnusedLayers` as well)

Reply:
I am not sure if I follow. What should we clear in `DetachUnusedLayers`?

Reply:
I asking if we should clear `current_composition_params[view_id]` when we remove `view_id` from the UIView tree in `DetachUnusedLayer` (e.g if we are re-attaching that view with the same composition parameters, do we need to call setframe again?)

Reply:
The frame of the subview won't change after removing it from superview. So if we remove the platform view in `DetachUnusedLayer` from the superview and the composite param doesn't change, it is ok to not call setFrame again.